src/runtime/rpc/network.cpp

```diff
@@ -391,13 +415,36 @@ bool rpc_session::on_disconnected(bool is_write)
     return ret;
 }
 
+bool rpc_session::is_auth_success(message_ex *msg)
+{
+    if (security::FLAGS_enable_auth && !_negotiation->negotiation_succeed()) {
+        dwarn_f("reject message({}) from {}, session {} client",
+                msg->rpc_code().to_string(),
+                _remote_addr.to_string(),
+                is_client() ? "is" : "isn't");
+        // reply response with ERR_UNAUTHENTICATED if msg is request
+        if (msg->header->context.u.is_request) {
+            _net.engine()->reply(msg->create_response(), ERR_UNAUTHENTICATED);
+        }
```

Comment:
This is only a 'check' type function, I think reply message in `on_recv_message ` would be better.